### PR TITLE
Avoid collisions https://github.com/pittmesh/ip-calculator/issues/25

### PR DESCRIPTION
**Pull request by jasonkhanlar**

### CODE REVIEW

File: mm-mac2ipv4.bash [hunk 1]
```diff
@@ -23,7 +23,7 @@ function list-all {
   for ip2 in `seq 0 255`;do
     mac4=$(printf "%02X\n" $ip2)
 
-    ((ip2=ip2%64 + 64))
+    ((ip2=ip2 % 32 + 96))
```
Comment (colindean) [inline mm-mac2ipv4.bash]: I didn't catch this in the original, but it's almost never a good idea to modify a variable which is the element of a collection under iteration. Many languages allow it, but it's a bad habit that too often leads to bugs.

I'll let it through, but please consider changing it.

File: mm-mac2ipv4.bash [hunk 2]
```diff
@@ -99,6 +99,7 @@ ip2=$(printf "%d" "0x$mac4")
 ip3=$(printf "%d" "0x$mac5")
 ip4=$(printf "%d" "0x$mac6")
 
-((ip2=ip2%64 + 64))
+((ip2=ip2 % 32 + 96))
```
Comment (colindean) [inline mm-mac2ipv4.bash]: Second time I've seen this particular math. Is it worth extracting to a function?

File: mm-mac2ipv4.tests [hunk 3]
```diff
@@ -40,24 +40,38 @@ rand0to255() {
...
-    done
+    # 3rd IPv4 octet
+    rand0to255
+    randMA[0]="${randMA[0]}:$(printf "%02X\n" $randdec)"
+    randMA[1]="${randMA[1]}:$(printf "%02X\n" $randdec)"
+    randMA[2]="${randMA[2]}:$(printf "%02X\n" $randdec)"
+    randMA[3]="${randMA[3]}:$(printf "%02X\n" $randdec)"
+    randIP="$randIP.$randdec"
+    # 4th IPv4 octet
+    rand0to255eo32 # every other 32
```
Comment (colindean) [inline mm-mac2ipv4.tests]: If you have to write a comment about a function name, you probably could have a better function name.

Comment (colindean) [inline mm-mac2ipv4.tests]: Also, are you not doing anything with this return value? Or is there a side effect that I'm missing?

**Comment by colindean** [comment]
I'm a little concerned with the unhandled return values, so I'm going to hold off on merging it until you've clarified those.

**Comment by Protagonistics** [comment]
Colin, I'm not certain I'm following your reasoning? This seems to solve
the collision problem.

**Comment by colindean** [comment]
Yes, but it’s not clear how.

**Comment by Protagonistics** [comment]
Darn right. The script works! Are there collisions? It seems it works.
Let's talk about it if it seems problematic.
On Jun 16, 2016 1:38 AM, "Colin Dean" notifications@github.com wrote:

Yes, but it’s not clear how.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
https://github.com/pittmesh/ip-calculator/pull/26#issuecomment-226392794,
or mute the thread
https://github.com/notifications/unsubscribe/ABuiCH44Hhq2XnM9ixVmPT_axCo0G9wLks5qMOFmgaJpZM4I2yQK
.

**Comment by colindean** [comment]
@Protagonistics, there are potential issues that @jasonkhanlar needs to address before we can merge this. Five to ten minutes of work probably.

**Comment by Protagonistics** [comment]
What are the potential issues? I'm guessing  you mean collisions?

**Comment by colindean** [comment]
Excellent refactoring!

**Pull request merged**